`make clippy` fails locally but not on CI

Currently, `make clippy` (as well as `cargo build` in the workspace root) is failing locally due to some [recent changes to `miden-base`](https://github.com/0xPolygonMiden/miden-base/commit/781ab4375c06938adbd32762f943cd55f241d3f7). Here is the error:
```
error[E0599]: no variant or associated item named `Local` found for enum `miden_objects::notes::NoteExecutionHint` in the current scope
  --> bin/faucet/src/handlers.rs:72:83
   |
72 | ...NoteExecutionHint::Local)
   |                       ^^^^^ variant or associated item not found in `NoteExecutionHint`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `miden-faucet` (bin "miden-faucet") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

The CI (which runs `make clippy`) however [has been passing](https://github.com/0xPolygonMiden/miden-node/pull/430/checks) which could be an issue. This could be related to the new caching action (though I'm not too familiar with how it works).

--- a/Makefile
+++ b/Makefile
@@ -100,7 +100,7 @@ install-faucet: ## Installs faucet
 
 .PHONY: install-remote-prover
 install-remote-prover: ## Install remote prover's CLI
-	$(BUILD_PROTO) cargo install --path bin/remote-prover --bin miden-remote-prover --features concurrent
+	$(BUILD_PROTO) cargo install --path bin/remote-prover --bin miden-remote-prover --features concurrent --locked
 
 .PHONY: install-stress-test
 install-stress-test: ## Installs stress-test binary

